packages/api/tests/models/rules/test-sns-rules.js

```diff
@@ -0,0 +1,316 @@
+
+  t.is(rule.rule.value, snsTopicArn);
+  t.falsy(rule.rule.arn);
+  t.is(rule.state, 'DISABLED');
+
+  const updates = {
+    name: rule.name,
+    state: 'ENABLED',
+    rule: {
+      ...rule.rule,
+      arn: 'test-value'
+    }
+  };
+
+  // Should fail because a disabled rule should not have an ARN
+  // when being updated
+  await t.throwsAsync(rulesModel.update(rule, updates), null,
+    'Including rule.arn is not allowed when enabling a disabled rule');
+  t.teardown(async () => {
+    await rulesModel.delete(rule);
+    snsStub.restore();
+  });
+});
+
+test.serial('updating an SNS rule updates the event source mapping', async (t) => {
+  const snsTopicArn = randomString();
+  const newSnsTopicArn = randomString();
+
+  const snsStub = sinon.stub(awsServices, 'sns')
+    .returns({
+      listSubscriptionsByTopic: () => ({
+        promise: () => Promise.resolve({
+          Subscriptions: [{
+            Endpoint: process.env.messageConsumer,
+            SubscriptionArn: randomString()
+          }]
+        })
+      }),
+      unsubscribe: () => ({
+        promise: () => Promise.resolve()
+      })
+    });
+
+  const item = fakeRuleFactoryV2({
+    workflow,
+    rule: {
+      type: 'sns',
+      value: snsTopicArn
+    },
+    state: 'ENABLED'
+  });
+
+  const rule = await rulesModel.create(item);
+
+  t.is(rule.rule.value, snsTopicArn);
+
+  const updates = { name: rule.name, rule: { value: newSnsTopicArn } };
+  const updatedRule = await rulesModel.update(rule, updates);
+
+  t.is(updatedRule.name, rule.name);
+  t.is(updatedRule.type, rule.type);
+  t.is(updatedRule.rule.value, newSnsTopicArn);
+  t.not(updatedRule.rule.arn, rule.rule.arn);
+
+  await rulesModel.delete(rule);
+  t.teardown(() => snsStub.restore());
+});
+
+test.serial('deleting an SNS rule updates the event source mapping', async (t) => {
+  const snsTopicArn = randomString();
+
+  const snsStub = sinon.stub(awsServices, 'sns')
+    .returns({
+      listSubscriptionsByTopic: () => ({
+        promise: () => Promise.resolve({
+          Subscriptions: [{
+            Endpoint: process.env.messageConsumer,
+            SubscriptionArn: randomString()
+          }]
+        })
+      }),
+      unsubscribe: () => ({
+        promise: () => Promise.resolve()
+      })
+    });
+  const unsubscribeSpy = sinon.spy(awsServices.sns(), 'unsubscribe');
+
+  const item = fakeRuleFactoryV2({
+    workflow,
+    rule: {
+      type: 'sns',
+      value: snsTopicArn
+    },
+    state: 'ENABLED'
+  });
+
+  const rule = await rulesModel.create(item);
+
+  t.is(rule.rule.value, snsTopicArn);
+
+  await rulesModel.delete(rule);
+
+  t.true(unsubscribeSpy.called);
+  t.true(unsubscribeSpy.calledWith({
+    SubscriptionArn: rule.rule.arn
+  }));
+
+  t.teardown(() => {
+    snsStub.restore();
+    unsubscribeSpy.restore();
+  });
+});
+
+test.serial('multiple rules using same SNS topic can be created and deleted', async (t) => {
```

Comment:
This is the test I added to confirm that we can create/delete multiple SNS rules using the same topic. The rest of the tests in here were moved from `test-rules-model.js` just to keep things more organized logically